scripts/maint/practracker/metrics.py

```diff
@@ -16,21 +20,28 @@ def get_include_count(f):
             include_count += 1
     return include_count
 
-def function_lines(f):
+def get_function_lines(f):
     """
     Return iterator which iterates over functions and returns (function name, function lines)
     """
 
-    # XXX Buggy! Doesn't work with MOCK_IMPL and ENABLE_GCC_WARNINGS
+    # Skip lines with these terms since they confuse our regexp
+    REGEXP_CONFUSE_TERMS = ["MOCK_IMPL", "ENABLE_GCC_WARNINGS", "ENABLE_GCC_WARNING", "DUMMY_TYPECHECK_INSTANCE",
+                            "DISABLE_GCC_WARNING", "DISABLE_GCC_WARNINGS"]
+
     in_function = False
     for lineno, line in enumerate(f):
+        if any(x in line for x in REGEXP_CONFUSE_TERMS):
```

Comment:
Instead of checking for the presence of the strings here, I'd suggest checking after we find the start of a function, to see if the function name is in the list of confusing terms.

Reply:
Added to post-merge ticket #29746.

Reply:
I think this one is actually important for correctness; I'll try a quick fix.  See e968b886aca66a in my bug29221_more